examples/core-auth-nextjs-pages-nextauth-4/src/pages/auth/signin.tsx

```diff
@@ -0,0 +1,96 @@
+import * as React from 'react';
+import type { GetServerSidePropsContext, InferGetServerSidePropsType } from 'next';
+import Link from '@mui/material/Link';
+import { SignInPage } from '@toolpad/core/SignInPage';
+import { getProviders, signIn } from 'next-auth/react';
+import { getServerSession } from 'next-auth/next';
+import { useRouter } from 'next/router';
+import type { AuthProvider } from '@toolpad/core';
+import { authOptions } from '../api/auth/[...nextauth]';
+
+function ForgotPasswordLink() {
+  return <Link href="/auth/forgot-password">Forgot password?</Link>;
+}
+
+function SignUpLink() {
+  return <Link href="/auth/signup">Sign up</Link>;
+}
+
+export default function SignIn({
+  providers,
+}: InferGetServerSidePropsType<typeof getServerSideProps>) {
+  const router = useRouter();
+  return (
+    <SignInPage
+      providers={providers}
+      signIn={async (provider, formData, callbackUrl) => {
```

Comment:
I would separate this function from the rendering not to mix up the logic with the view so much, makes it more readable. Just personal preference though!